src/database/statistics/dateUtil.ts

```diff
@@ -1,76 +1,47 @@
 import { DevSpreadDates } from 'src/github-api/model/DevFocus';
 
 /**
- * Helper function to eliminate duplicate weeks, sprints or months
- * because the calculated intervals per developer may differ
- * and would have been taken as separate intervals into account,
- * altough it was the same interval of commiting.
- * @param timeSpreadPairs An object with timestamps as keys, and an object with developer:sprad pairs as a value
- * @param days The days, the interval blongs to, i.e. 7 (weeks), 14 (sprint), 30(month)
- * @returns
+ * Computes all spread values for a developer,
+ * based on his @param timeRepoPairs. The timestamp
+ * is the commit date, the repo Ids identify the repos,
+ * in which was committed on that date.
+ * @param timeRepoPairs timestamp: [repoId] pairs object.
+ * @returns An object with computed spreads for time category,
+ * the sum of the spread for each time category
+ * and the total number of timestamps in a category (days, weeks, ...)
  */
-export function rearangeTimeslots(
-  timeSpreadPairs: { [key: string]: { [key: string]: number } },
-  days: number,
-): { [key: string]: { [key: string]: number } } {
-  // copy the old timeSpreadPairs to modify on that
-  const newTimeSpreadPairs: { [key: string]: { [key: string]: number } } = {
-    ...timeSpreadPairs,
-  };
-  const timestamps: string[] = Object.keys(timeSpreadPairs).sort();
-  // go through every sorted timestamp and look to successor
-  for (let i = 1; i <= timestamps.length; ) {
-    const currentDate: string = timestamps[i - 1];
-    const nextDate: string = timestamps[i];
-    // check, if the next interval date should actualy be in current interval
-    if (addDays(currentDate, days) > new Date(timestamps[i])) {
-      // append the values together, update the current interval object
-      const devSpreadObj1 = timeSpreadPairs[currentDate];
-      const devSpreadObj2 = timeSpreadPairs[nextDate];
-      const mergedObj = { ...devSpreadObj1, ...devSpreadObj2 };
-      newTimeSpreadPairs[currentDate] = mergedObj;
-      // delete the unnessacary date and skip the next date
-      delete newTimeSpreadPairs[nextDate];
-      i += 2;
-    } else {
-      i += 1;
-    }
-  }
-  return newTimeSpreadPairs;
-}
+export function getSpreadsForDev(timeRepoPairs: {
+  [key: string]: string[];
+}): DevSpreadDates {
+  // sort because timestamp order is broken after mixing them together from different repos
+  const timestamps: string[] = Object.keys(timeRepoPairs).sort();
```

Comment:
Just a heads up: JavaScript sorts in [weird ways](https://developer.mozilla.org/de/docs/Web/JavaScript/Reference/Global_Objects/Array/sort) based on UTF-16.

To be safe, you should take a look at [lodash's sortBy method](https://lodash.com/docs/4.17.15#sortBy)

Reply:
It seems to work with the standard library sort, but I'll definitely keep this in mind when I'll do the unit tests.